Generator/Sources/NeedleFramework/Generating/DependencyProviderSerializerTask.swift

```diff
@@ -34,28 +34,60 @@ class DependencyProviderSerializerTask: AbstractTask<[SerializedProvider]> {
     ///
     /// - returns: The list of `SerializedProvider`.
     override func execute() -> [SerializedProvider] {
-        return providers.map { (provider: ProcessedDependencyProvider) in
-            return serialize(provider)
+        var result = [SerializedProvider]()
+        // Group the providers based on where the properties are coming from
+        // This will allow us to extract common code for multiple depndency providers
+        // into common base classes
+        var counts = [[ProcessedProperty]: [ProcessedDependencyProvider]]()
+        for provider in providers {
+            let properties = provider.processedProperties
+            counts[properties, default: []].append(provider)
         }
+        let countValues = counts.values.sorted(by: { (left, right) in
```

Comment:
I suppose you could import `TSCBasic` in this file and use its `OrderedDictionary`? Seems like it'd be faster and cleaner than performing the sort.

Reply:
I *think* either implementation would solve the problem of "different needle file in consecutive runs". Have you verified that?